internal/config/migration/migration.go

```diff
@@ -9,305 +9,12 @@ package migration
-			cfgV3.Attestation.AzureSEVSNP.FirmwareSignerConfig = config.SNPFirmwareSignerConfig{
-				AcceptedKeyDigests: cfgV2.Provider.Azure.IDKeyDigest,
-				EnforcementPolicy:  idkeydigest.Enforcement(cfgV2.Provider.Azure.EnforceIDKeyDigest),
-			}
-		}
-	case cfgV2.Provider.GCP != nil:
-		cfgV3.Provider.GCP = &config.GCPConfig{
-			Project:               cfgV2.Provider.GCP.Project,
-			Region:                cfgV2.Provider.GCP.Region,
-			Zone:                  cfgV2.Provider.GCP.Zone,
-			ServiceAccountKeyPath: cfgV2.Provider.GCP.ServiceAccountKeyPath,
-			InstanceType:          cfgV2.Provider.GCP.InstanceType,
-			StateDiskType:         cfgV2.Provider.GCP.StateDiskType,
-			DeployCSIDriver:       cfgV2.Provider.GCP.DeployCSIDriver,
-		}
-		cfgV3.Attestation.GCPSEVES = &config.GCPSEVES{
-			Measurements: cfgV2.Provider.GCP.Measurements,
-		}
-	case cfgV2.Provider.OpenStack != nil:
-		cfgV3.Provider.OpenStack = &config.OpenStackConfig{
-			Cloud:             cfgV2.Provider.OpenStack.Cloud,
-			AvailabilityZone:  cfgV2.Provider.OpenStack.AvailabilityZone,
-			FlavorID:          cfgV2.Provider.OpenStack.FlavorID,
-			FloatingIPPoolID:  cfgV2.Provider.OpenStack.FloatingIPPoolID,
-			AuthURL:           cfgV2.Provider.OpenStack.AuthURL,
-			ProjectID:         cfgV2.Provider.OpenStack.ProjectID,
-			ProjectName:       cfgV2.Provider.OpenStack.ProjectName,
-			UserDomainName:    cfgV2.Provider.OpenStack.UserDomainName,
-			ProjectDomainName: cfgV2.Provider.OpenStack.ProjectDomainName,
-			RegionName:        cfgV2.Provider.OpenStack.RegionName,
-			Username:          cfgV2.Provider.OpenStack.Username,
-			Password:          cfgV2.Provider.OpenStack.Password,
-			DirectDownload:    cfgV2.Provider.OpenStack.DirectDownload,
-		}
-		cfgV3.Attestation.QEMUVTPM = &config.QEMUVTPM{
-			Measurements: cfgV2.Provider.OpenStack.Measurements,
-		}
-	case cfgV2.Provider.QEMU != nil:
-		cfgV3.Provider.QEMU = &config.QEMUConfig{
-			ImageFormat:           cfgV2.Provider.QEMU.ImageFormat,
-			VCPUs:                 cfgV2.Provider.QEMU.VCPUs,
-			Memory:                cfgV2.Provider.QEMU.Memory,
-			MetadataAPIImage:      cfgV2.Provider.QEMU.MetadataAPIImage,
-			LibvirtURI:            cfgV2.Provider.QEMU.LibvirtURI,
-			LibvirtContainerImage: cfgV2.Provider.QEMU.LibvirtContainerImage,
-			NVRAM:                 cfgV2.Provider.QEMU.NVRAM,
-			Firmware:              cfgV2.Provider.QEMU.Firmware,
-		}
-		cfgV3.Attestation.QEMUVTPM = &config.QEMUVTPM{
-			Measurements: cfgV2.Provider.QEMU.Measurements,
-		}
-	}
-
-	// Create backup
-	if err := os.Rename(path, path+".backup.v2"); err != nil {
-		return fmt.Errorf("creating backup: %w", err)
-	}
-
-	// Write migrated config
-	if err := fileHandler.WriteYAML(path, cfgV3, file.OptOverwrite); err != nil {
-		return fmt.Errorf("writing %s: %w", path, err)
-	}
-
-	return nil
+func V2ToV3(_ string, _ file.Handler) error {
+	// TODO(malt3): add migration from v3 to v4
```

Comment:
You are planning to add this in a separate PR?

Reply:
Yes. It was actually part of this PR and was moved out to make this PR more readable.